Keda.Cosmosdb.Scaler/src/Services/CosmosDBExternalScaler.cs

```diff
@@ -0,0 +1,173 @@
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Logging;
+using static Keda.CosmosDB.Scaler.Protos.ExternalScaler;
+
+namespace Keda.CosmosDB.Scaler.Services
+{
+    public class CosmosDBExternalScaler : ExternalScalerBase
+    {
+        private readonly ILogger _logger;
+
+        public CosmosDBExternalScaler(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<CosmosDBExternalScaler>();
+        }
+
+        public override async Task<IsActiveResponse> IsActive(ScaledObjectRef request, ServerCallContext context)
+        {
+            try
+            {
+                var trigger = CreateTriggerFromMetadata(request.ScalerMetadata);
+                var workToBeDone = await GetEstimatedWork(trigger);
+
+                bool isActive = workToBeDone > 0;
+
+                if (isActive)
+                {
+                    _logger.LogDebug(string.Format("Activating to 1 for cosmosDB account {0} database {1} collection {2}", trigger.AccountName, trigger.DatabaseName, trigger.CollectionName));
+                }
+
+                return new IsActiveResponse
+                {
+                    Result = isActive
+                };
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error getting estimated work.");
+                throw ex;
+            }
+        }
+
+        public override Task<GetMetricSpecResponse> GetMetricSpec(ScaledObjectRef request, ServerCallContext context)
+        {
+            var resp = new GetMetricSpecResponse();
+            resp.MetricSpecs.Add(new MetricSpec
+            {
+                MetricName = NormalizeString(string.Format("{0}-{1}-{2}-{3}", "azure-cosmosDB",
+                                        request.ScalerMetadata[CosmosDBTriggerMetadata.AccountName],
+                                        request.ScalerMetadata[CosmosDBTriggerMetadata.DatabaseName],
+                                        request.ScalerMetadata[CosmosDBTriggerMetadata.CollectionName])),
+                TargetSize = 1
+            });
+            return Task.FromResult(resp);
+        }
+
+        public override async Task<GetMetricsResponse> GetMetrics(GetMetricsRequest request, ServerCallContext context)
+        {
+            var resp = new GetMetricsResponse();
+            try
+            {
+                var trigger = CreateTriggerFromMetadata(request.ScaledObjectRef.ScalerMetadata);
+                long workToBeDone = await GetEstimatedWork(trigger);
+
+                var metricName = request.ScaledObjectRef.ScalerMetadata["MetricName"];
+                resp.MetricValues.Add(new MetricValue
+                {
+                    MetricName = metricName,
+                    MetricValue_ = workToBeDone
+                });
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error generating GetMetricsResponse");

```

Comment:
Same here, leave out the log to avoid multiple entries?

Reply:
See above